tests/unit/test_core_drain.py

```
@@ -0,0 +1,399 @@
+# ---------------------------------------------------------------------------
+
+
+def test_token_is_frozen_dataclass() -> None:
+    """``_TransportOperationToken`` is frozen — value equality, not identity.
+
+    Locked down so a follow-up refactor doesn't accidentally turn this into
+    a mutable dataclass, which would break dict-keying and reduce safety
+    around begin/finish pairing.
+    """
+    token = _TransportOperationToken(task=None)
+    with pytest.raises(Exception):  # FrozenInstanceError subclasses Exception
```

Ruff (0.15.12)

[warning] Do not assert blind exception: `Exception`

(B017)